src/plugins/saved_search/public/plugin.ts

```diff
@@ -147,3 +153,15 @@ export class SavedSearchPublicPlugin
     };
   }
 }
+
+registerSavedObjectToPanelMethod(SavedSearchType, (savedObject) => {
+  if (!savedObject.managed) {
+    return { savedObjectId: savedObject.id } as SavedObjectEmbeddableInput;
```

Comment:
Is the `as SavedObjectEmbeddableInput` portion needed here? I'm not very familiar with these types but it looks like it is missing some properties from that type spec.

Reply:
I'm open for removing it, but I left it in to clarify where this structure is coming from. It's true that it's a partial. But that's what is expected by the `container.addNewEmbeddable` method (see [this line](https://github.com/elastic/kibana/blob/main/src/plugins/embeddable/public/lib/containers/container.ts#L189), `explicitInput` parameter).

I could always say `as Partial<SavedObjectEmbeddableInput>`. Do you think that would be better?

Reply:
Maybe updating the types in `saved_object_to_panel_methods` could help make this clearer? From what I can tell it looks like we're expecting to receive raw SO attributes and return either an SO ID if it's unmanaged or by value attributes if it's managed. Rather than making the return type `any` and having the consumers know what to cast to, we could instead make `SavedObjectToPanelMethod` generic and have consumers explicitly type it when they register their callback. Something like this is what I'm thinking:
```ts
// saved_object_to_panel_methods.ts
type SavedObjectToPanelMethod<TSavedObjectAttributes, TByValueAttributes> = (
  savedObject: SavedObjectCommon<TSavedObjectAttributes>
) => { savedObjectId: string } | { attributes: TByValueAttributes };

export const savedObjectToPanel: Record<string, SavedObjectToPanelMethod<any, any>> = {};

export const registerSavedObjectToPanelMethod = <TSavedObjectAttributes, TByValueAttributes>(
  savedObjectType: string,
  method: SavedObjectToPanelMethod<TSavedObjectAttributes, TByValueAttributes>
) => {
  savedObjectToPanel[savedObjectType] = method;
};

// plugin.ts
registerSavedObjectToPanelMethod<SavedSearchAttributes, SavedSearchByValueAttributes>(
  SavedSearchType,
  (savedObject) => {
    if (!savedObject.managed) {
      return { savedObjectId: savedObject.id };
    }

    return {
      attributes: savedObjectToEmbeddableAttributes(savedObject),
    };
  }
);
```

I gave it a quick shot locally to see if it would work, and it did other than a type error in `add_panel_flyout.tsx`. But I think that's because there's actually a type issue there, and it might still be best to keep the casting internal to the embeddable plugin if needed.

Reply:
Done!